pkg/kvstore/etcd.go

```diff
@@ -151,25 +151,23 @@ type etcdClient struct {
 	// is set up in the etcd Client.
 	firstSession chan struct{}
 
-	client *client.Client
```

Comment:
Is this clean up critical? It will make backports more difficult. Please split out the non-critical parts.

Reply:
I think so because it was confusing for me to track down what mutex was protecting what.